libraries/app/api.cpp

```diff
@@ -328,6 +328,10 @@ namespace graphene { namespace app {
              start = node.operation_id;
        } catch(...) { return result; }
 
+       auto es = _app.get_plugin<elasticsearch::elasticsearch_plugin>("elasticsearch");
+       if(es)
+          return es->get_account_history(account, stop, limit, start);
```

Comment:
Problem: this works synchronously, inside the database thread. It is important to decouple this.
The ES query plugin should set up a separate thread for executing the queries and parsing the responses. The database thread should then make the call to the ES thread and wait for the result to return. Waiting will `yield`, which frees up the DB thread for other things.

Reply:
I understand the importance of this.

Can you point me in the right direction on how to make it following the graphene/bitshares style. Example in the codebase, functions to use, etc will be of great help.

Thank you.

Reply:
We don't have many examples where threading is used.

Check the network node.
* create a thread: https://github.com/bitshares/bitshares-core/blob/3.0.1/libraries/net/node.cpp#L285
* execute something in the other thread: https://github.com/bitshares/bitshares-core/blob/3.0.1/libraries/net/node.cpp#L4641

`_async` returns a future. Waiting for the future result will yield, i. e. other tasks within the current thread will be executed in parallel. Not sure what that means in the context of an API call, you'll have to experiment there.